src/test/java/io/confluent/examples/streams/StreamToStreamJoinIntegrationTest.java

```diff
@@ -104,48 +99,52 @@ public void shouldJoinTwoStreams() throws Exception {
     // for the same join key (e.g. "newspaper-advertisement"), we receive an update from either of
     // the two joined streams during the defined join window.
     final KStream<String, String> impressionsAndClicks = alerts.outerJoin(incidents,
-        (impressionValue, clickValue) -> impressionValue + "/" + clickValue,
-        // KStream-KStream joins are always windowed joins, hence we must provide a join window.
-        JoinWindows.of(TimeUnit.SECONDS.toMillis(5)));
+      (impressionValue, clickValue) -> impressionValue + "/" + clickValue,
+      // KStream-KStream joins are always windowed joins, hence we must provide a join window.
+      JoinWindows.of(TimeUnit.SECONDS.toMillis(5)));
 
     // Write the results to the output topic.
     impressionsAndClicks.to(outputTopic);
 
     final TopologyTestDriver topologyTestDriver = new TopologyTestDriver(builder.build(), streamsConfiguration);
 
-    //
-    // Step 2: Publish ad impressions.
-    //
-    IntegrationTestUtils.produceKeyValuesSynchronously(
-      adImpressionsTopic,
-      inputAdImpressions,
-      topologyTestDriver,
-      new StringSerializer(),
-      new StringSerializer()
-    );
-
-    //
-    // Step 3: Publish ad clicks.
-    //
-    IntegrationTestUtils.produceKeyValuesSynchronously(
-      adClicksTopic,
-      inputAdClicks,
-      topologyTestDriver,
-      new StringSerializer(),
-      new StringSerializer()
-    );
+    try {
+      //
+      // Step 2: Publish ad impressions.
+      //
+      IntegrationTestUtils.produceKeyValuesSynchronously(
+        adImpressionsTopic,
+        inputAdImpressions,
+        topologyTestDriver,
+        new StringSerializer(),
+        new StringSerializer()
+      );
 
-    //
-    // Step 4: Verify the application's output data.
-    //
-    final List<KeyValue<String, String>> actualResults =
-      IntegrationTestUtils.drainStreamOutput(
-        outputTopic,
+      //
+      // Step 3: Publish ad clicks.
+      //
+      IntegrationTestUtils.produceKeyValuesSynchronously(
+        adClicksTopic,
+        inputAdClicks,
         topologyTestDriver,
-        new StringDeserializer(),
-        new StringDeserializer()
+        new StringSerializer(),
+        new StringSerializer()
       );
-    assertThat(actualResults).containsExactlyElementsOf(expectedResults);
+
+      //
+      // Step 4: Verify the application's output data.
+      //
+      final List<KeyValue<String, String>> actualResults =
+        IntegrationTestUtils.drainStreamOutput(
+          outputTopic,
+          topologyTestDriver,
+          new StringDeserializer(),
+          new StringDeserializer()
+        );
+      assertThat(actualResults).containsExactlyElementsOf(expectedResults);
```

Comment:
Seems we refactored some example to use map instead of key-value list. Do we want to do this for all similar use cases?

Reply:
no, only for the tabular examples. In this case, we're producing a stream, so we're verifying the stream of updates is as expected.